omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/ConceptClassController1_8Test.java

```diff
@@ -93,6 +95,25 @@ public void shouldRetireAConceptClass() throws Exception {
 		Assert.assertEquals(reason, service.getConceptClassByUuid(uuid).getRetireReason());
 	}
 	
+	@Test
+	public void shouldUnRetireAConceptClass() throws Exception {
+		ConceptClass conceptClass = service.getConceptClassByUuid(getUuid());
+		conceptClass.setRetired(true);
+		conceptClass.setRetiredBy(Context.getAuthenticatedUser());
+		conceptClass.setDateRetired(new Date());
+		conceptClass.setRetireReason("random reason");
+		service.saveConceptClass(conceptClass);
+		Assert.assertTrue(conceptClass.isRetired());
```

Comment:
Will the above assertion fail if line 105 is commented out?

Reply:
Definitely not, what was in my mind though was `conceptClass = service.saveConceptClass(conceptClass);`. I am refactoring to reflect this.